components/backup_text/src/hr_datum.rs

```diff
@@ -311,6 +208,11 @@ impl From<Datum> for HrDatum {
             Datum::Max => Self::Max,
         }
     }
+
+    pub fn with_workload_sim_mask(d: Datum, field_type: Option<&dyn FieldTypeAccessor>) -> Self {
```

Comment:
```suggestion
    pub fn with_workload_sim_mask(d: Datum, field_type: &dyn FieldTypeAccessor) -> Self {
```
The `field_type` can't be `None`, or maybe we can not use the `field_type` at all. Since currently `field_type` is only used for resizing number values, we can first get the scope (u8, u16, etc) where the value falls in then resize the hashed value to that scope. Although the scope may not be the same as its actual type (will be smaller), it is okay as long as we keep the workload data hashing in the same way.

Reply:
This seems like a good idea.